File: summarize-lighthouse.mjs
import fs from 'node:fs';
import path from 'node:path';

const dir = '/home/ubuntu/crest-automotive-astro/lighthouse-reports';
const files = fs.readdirSync(dir).filter((file) => file.endsWith('.json') && file !== 'summary.json' && file !== 'summary-output.json').sort();
const rows = [];
const auditFindings = new Map();

for (const file of files) {
  const report = JSON.parse(fs.readFileSync(path.join(dir, file), 'utf8'));
  const scores = Object.fromEntries(Object.entries(report.categories || {}).map(([key, value]) => [key, value.score == null ? null : Math.round(value.score * 100)]));
  const metrics = Object.fromEntries(['first-contentful-paint','largest-contentful-paint','speed-index','total-blocking-time','cumulative-layout-shift','interactive'].map((id) => [id, report.audits?.[id]?.displayValue || report.audits?.[id]?.numericValue || null]));
  rows.push({ file, scores, metrics });
  for (const [id, audit] of Object.entries(report.audits || {})) {
    const isRelevant = ['accessibility','seo','best-practices','performance'].some((category) => report.categories?.[category]?.auditRefs?.some((ref) => ref.id === id));
    if (!isRelevant || audit.score == null || audit.score >= 0.9 || audit.scoreDisplayMode === 'notApplicable' || audit.scoreDisplayMode === 'informative') continue;
    if (!auditFindings.has(id)) auditFindings.set(id, { title: audit.title, description: audit.description, routes: [] });
    auditFindings.get(id).routes.push({ file, score: audit.score, displayValue: audit.displayValue || '', details: audit.details?.items?.slice?.(0, 3) || [] });
  }
}

const summary = { generatedAt: new Date().toISOString(), reports: rows, findings: Object.fromEntries(auditFindings) };
fs.writeFileSync(path.join(dir, 'summary.json'), JSON.stringify(summary, null, 2));
console.log(JSON.stringify(summary, null, 2));

File: scripts/summarize_lighthouse.py
from pathlib import Path
import json

report_dir = Path('/home/ubuntu/crest-automotive-astro/lighthouse-reports/final')
out = ['# Final Lighthouse summary\n']
for path in sorted(report_dir.glob('*.json')):
    data = json.loads(path.read_text())
    out.append(f'## {path.stem}\n')
    for key in ('performance', 'accessibility', 'best-practices', 'seo'):
        score = data.get('categories', {}).get(key, {}).get('score')
        out.append(f'- {key}: {score * 100:.0f}' if score is not None else f'- {key}: unavailable')
    failed = []
    for audit_id, audit in data.get('audits', {}).items():
        if audit.get('score') is not None and audit.get('score') < 1 and audit.get('scoreDisplayMode') not in {'notApplicable', 'manual', 'informative'}:
            failed.append((audit_id, audit.get('title', audit_id), audit.get('score')))
    if failed:
        out.append('\nPotential follow-ups:')
        for audit_id, title, score in sorted(failed, key=lambda item: item[2]):
            out.append(f'- `{audit_id}` — {title} ({score})')
    else:
        out.append('\nNo failing Lighthouse audits in the selected categories.')
    out.append('')
Path('/home/ubuntu/crest-automotive-astro/lighthouse-reports/final/summary.md').write_text('\n'.join(out))
print('\n'.join(out))
